File: frontend/src/types/image.ts
import { GeoJSONGeometry } from './index';

export type ImageProcessingStatus = 'UPLOADED' | 'QUEUED' | 'PROCESSING' | 'COMPLETED' | 'FAILED';

export interface DroneImage {
  id: string;
  mission_id: string;
  farm_id?: string;
  zone_id?: string;
  filename: string;
  file_type: string;
  file_size_bytes: number;
  sensor_type: string;
  image_type: string;
  processing_status: ImageProcessingStatus;
  capture_time: string;
  altitude_agl_meters: number;
  location?: GeoJSONGeometry;
  footprint?: GeoJSONGeometry;
  created_at: string;
}

export interface DroneImageListResponse {
  images: DroneImage[];
  total: number;
}

export interface ImageUploadResponse {
  image_id: string;
  filename: string;
  file_size_bytes: number;
  sensor_type: string;
  processing_status: ImageProcessingStatus;
  job_id: string;
  message: string;
}

export interface ImageProcessingJob {
  id: string;
  mission_id: string;
  job_type: string;
  status: 'queued' | 'running' | 'completed' | 'failed';
  progress_percent: number;
  celery_task_id?: string;
  output_artifacts?: Record<string, any>;
  error_message?: string;
  created_at: string;
}

File: backend/app/schemas/image.py
from datetime import datetime
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field

from app.schemas.common import BaseSchema


class DroneImageResponse(BaseSchema):
    """Metadata and spatial location of an ingested drone image frame."""

    id: str
    mission_id: str
    farm_id: Optional[str] = None
    zone_id: Optional[str] = None
    filename: str
    file_type: str
    file_size_bytes: int
    sensor_type: str
    image_type: str
    processing_status: str
    capture_time: datetime
    altitude_agl_meters: float
    location: Optional[Dict[str, Any]] = Field(
        default=None, description="GeoJSON Point GPS location (SRID 4326)"
    )
    footprint: Optional[Dict[str, Any]] = Field(
        default=None, description="GeoJSON Polygon ground footprint (SRID 4326)"
    )
    created_at: datetime


class DroneImageListResponse(BaseSchema):
    """Collection of drone imagery frames."""

    images: List[DroneImageResponse]
    total: int


class ImageUploadResponse(BaseSchema):
    """Immediate acknowledgement returned upon asynchronous image upload."""

    image_id: str
    filename: str
    file_size_bytes: int
    sensor_type: str
    processing_status: str
    job_id: str
    message: str


class ImageProcessingJobResponse(BaseSchema):
    """Status of an asynchronous background processing job."""

    id: str
    mission_id: str
    job_type: str
    status: str
    progress_percent: float
    celery_task_id: Optional[str] = None
    output_artifacts: Optional[Dict[str, Any]] = None
    error_message: Optional[str] = None
    created_at: datetime
